--- A/wifi/drivers/connmanctl.c	2018-09-08 05:52:59.000000000 +0200
+++ B/wifi/drivers/connmanctl.c	2018-11-11 16:11:20.000000000 +0100
@@ -205,6 +205,8 @@
    fclose(settings_file);
 
    pclose(popen("systemctl restart connman", "r"));
+   
+   pclose(popen("connmanctl agent on ", "r"));
 
    strlcat(command, "connmanctl connect ", sizeof(command));
    strlcat(command, service, sizeof(command));
